patches/server/1052-add-number-format-api.patch

```diff
@@ -0,0 +1,171 @@
++        }
++
++        throw new IllegalArgumentException("Unknown format type " + vanilla.getClass());
++    }
++}
+diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
+index 3157f3d2f9ce7af4a763203672817a7f5c7bd4fb..1cc3c43e4139fbf08b0b1e19e15cd92a51e6728e 100644
+--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
++++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
+@@ -152,6 +152,34 @@ final class CraftObjective extends CraftScoreboardComponent implements Objective
+     }
+     // Paper end
+ 
++    // Paper start - add number format
++    @Override
++    public io.papermc.paper.scoreboard.numbers.NumberFormat numberFormat() {
++        this.checkState();
++
++        net.minecraft.network.chat.numbers.NumberFormat vanilla = this.objective.numberFormat();
++        
++        if (vanilla == null) {
++            return null;
++        }
++
++        return io.papermc.paper.util.PaperScoreboardFormat.asPaper(vanilla);
++    }
++
++
++    @Override
++    public void numberFormat(io.papermc.paper.scoreboard.numbers.NumberFormat format) {
++        this.checkState();
++
++        if (format == null) {
++            this.objective.setNumberFormat(null);
++            return;
++        }
++
++        this.objective.setNumberFormat(io.papermc.paper.util.PaperScoreboardFormat.asVanilla(format));
++    }
++    // Paper end - add number format
++
+     @Override
+     public void unregister() {
+         CraftScoreboard scoreboard = this.checkState();
+diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftScore.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftScore.java
+index be020d6b686c1ad5bd8b7cee0b6050304d434022..a66bdeb8d96e4fe7de9738bb2be0a17a6bef5d09 100644
+--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftScore.java
++++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftScore.java
+@@ -57,6 +57,44 @@ final class CraftScore implements Score {
+         this.objective.checkState().board.getOrCreatePlayerScore(this.entry, this.objective.getHandle()).set(score);
+     }
+ 
++
++    // Paper start - add number format
++    @Override
++    public io.papermc.paper.scoreboard.numbers.NumberFormat numberFormat() {
++        Scoreboard board = this.objective.checkState().board;
++        if (board.getTrackedPlayers().contains(this.entry)) { // Lazy
++            ReadOnlyScoreInfo scoreInfo = this.objective.checkState().board
++                .getPlayerScoreInfo(this.entry, this.objective.getHandle());
++
++            if (scoreInfo != null) {
++                net.minecraft.network.chat.numbers.NumberFormat vanilla = scoreInfo.numberFormat();
++
++                if (vanilla == null) {
++                    return null;
++                }
++
++                return io.papermc.paper.util.PaperScoreboardFormat.asPaper(vanilla);
++            }
++        }
```

Comment:
This logic is wrong and is actually a bug fixed in another PR. Specifically the part about doing a `contains` check on `getTrackedPlayers`. Just use the null returned by getPlayerScoreInfo as a way of knowing if the score exists.